pkg/transformers/config/defaultconfig/namereference.go

```diff
@@ -299,10 +299,10 @@ nameReference:
 - kind: ServiceAccount
   version: v1
   fieldSpecs:
-  - path: subjects/name
+  - path: subjects
```

Comment:
Yes. This is need to stop the mutateField recursion at the right level. You can not mutate the field without having access to the namespace. Changing the namereference.go get your to enter that code: https://github.com/kubernetes-sigs/kustomize/pull/1325/files#diff-c9aa1ebf15951fe590c4241e95321901R228

Reply:
Added comment to explain the changes. This would help future contributor to understand the behavior: https://github.com/kubernetes-sigs/kustomize/pull/1325/commits/9b40f8ab474007a4f2358586bc82460b4bcac577